[PL-48906] Clock synchronization (#6254)
harness/developer-hub: docs/platform/settings/clock-sychronization.md

---
title: Clock synchronization
description: This topic explains recommended settings for clock synchronization.
sidebar_position: 2
---

This topic provides Harness' recommended settings for Network Time Protocol (NTP).

- **Control:** Clocks in all related information management systems should be integrated into a single reference time source for an organization.

- **Implementation:** A standard reference time should be defined for use inside the organization.

## Recommended NTP configuration for Harness Delegate

- Use your local NTP server
- Use [Google Public NTP](https://developers.google.com/time)

## Recommended NTP configuration for Harness Self-Managed Enterprise Edition

- Use your local NTP server

> - Use Use [Google Public NTP](https://developers.google.com/time)

Author: ```suggestion
- Use [Google Public NTP](https://developers.google.com/time)
```

## Configure NTP for a delegate running on Linux

Most Linux distributions use `systemd`, which comes with NTP for clock synchronization. You can verify `systemd` by running `timedatectl`.

### Install the NTP service

> 1. If NTP is not present on your host system, you can use `yum`, `apt-get`, or `dnf` to install the NTP service as per your OS. You can also use Chrony, which has flexible implementation of NTP.

Author: ```suggestion
1. If NTP is not present on your host system, you can use `yum`, `apt-get`, or `dnf` to install the NTP service as per your OS. You can also use Chrony, which has a flexible implementation of NTP.
```

- Use [Google Public NTP](https://developers.google.com/time): Google Public NTP presents a dependable option for time synchronization, delivering accurate time information across diverse network environments.

## Configure NTP for a delegate running on Linux

Most Linux distributions use `systemd`, which comes with NTP for clock synchronization. You can verify `systemd` by running `timedatectl`.

### Install the NTP service

1. If NTP is not present on your host system, you can use `yum`, `apt-get`, or `dnf` to install the NTP service as per your OS. You can also use Chrony, which has a flexible implementation of NTP.

   You must have root permissions to install the application. Installation varies slightly depending on which Linux distribution you use:

    ```
     sudo apt-get install ntp  # for Debian/Ubuntu
     sudo yum install ntp      # for CentOS/RHEL
     sudo dnf install ntp      # for Fedora
     ```

    NTP is configured using a `ntp.conf` configuration file. The file is generally located in the `/etc/` directory.

2. You can add multiple NTP servers in the `ntp.conf` file in the format below.

    :::info

>     The prefer option should only be specified once.

Reviewer: ```suggestion
    Add the `prefer` option to one `server` command only in the `ntp.conf` file.
```
